lib/IRGen/GenMeta.cpp

```diff
@@ -2432,6 +2463,27 @@ namespace {
 
     void noteStartOfImmediateMembers(ClassDecl *theClass) {}
 
+    void addValueWitnessTable() {
+      switch (IGM.getClassMetadataStrategy(Target)) {
+      case ClassMetadataStrategy::Resilient:
+      case ClassMetadataStrategy::Singleton:
+        // The runtime fills in the value witness table for us.
+        B.add(llvm::ConstantPointerNull::get(IGM.WitnessTablePtrTy));
```

Comment:
Seems redundant for the "singleton" case, no?

Reply:
I should add a comment. On Windows, we cannot emit an absolute reference to a symbol from another shared library. So the runtime has to fill in the value witness table, and we cannot use the fixed class metadata strategy at all. We could special-case this and still emit the value witness table statically on non-Windows but it doesn’t seem beneficial.